File: android/app/src/main/kotlin/in/nann/smashhockey/sketch/SketchData.kt
package `in`.nann.smashhockey.sketch

// THROWAWAY (SMASH-5): the motion sketch's fake data — the twin of iOS's SketchData.swift, a
// click-dummy for judging the 3D UI's feel on a phone, not the season (§11). Kit colours are
// invented here; the spec gives each club a primary and a secondary kit colour but does not yet
// name them.

enum class SketchScreenId { TITLE, HUB, MATCH, RESULT, COACH }

class SketchClub(val code: String, val primary: Int, val secondary: Int)

class SketchStanding(val code: String, val played: Int, val goalDiff: Int, val points: Int)

class SketchGoal(val at: Double, val home: Boolean)

object SketchData {
    val clubs: Map<String, SketchClub> = listOf(
        SketchClub("MOS", 0x4F8A3C, 0xF2E6C9),
        SketchClub("GLO", 0x6C4AB6, 0xFFD84D),
        SketchClub("NEB", 0x1F3B73, 0x7FD1F5),
        SketchClub("ROC", 0xE4572E, 0x2B2D42),
        SketchClub("DUN", 0xE8A33D, 0x5A2E0E),
        SketchClub("MIR", 0xC1476B, 0xF6E7D8),
        SketchClub("GLW", 0x9CC7E0, 0x1D3557),
        SketchClub("COR", 0xFF7F66, 0x0F6E7A),
    ).associateBy { it.code }
    const val PLAYER = "DUN"
    const val OPPONENT = "NEB"

    /** The table after matchday 4. */
    val before = listOf(
        SketchStanding("ROC", 4, 6, 10),
        SketchStanding("NEB", 4, 4, 9),
        SketchStanding("GLO", 4, 2, 7),
        SketchStanding("DUN", 4, 1, 6),
        SketchStanding("MOS", 4, 0, 5),
        SketchStanding("COR", 4, -2, 4),
        SketchStanding("MIR", 4, -4, 3),
        SketchStanding("GLW", 4, -7, 1),
    )

    /** The table after matchday 5: DUN 3–1 NEB, ROC 1–1 GLO, MOS 2–1 COR, MIR 1–0 GLW. */
    val after = ranked(listOf(
        SketchStanding("ROC", 5, 6, 11),
        SketchStanding("NEB", 5, 2, 9),
        SketchStanding("GLO", 5, 2, 8),
        SketchStanding("DUN", 5, 3, 9),
        SketchStanding("MOS", 5, 1, 8),
        SketchStanding("COR", 5, -3, 4),
        SketchStanding("MIR", 5, -3, 6),
        SketchStanding("GLW", 5, -8, 1),
    ))

    /** §11.4's order: points, goal difference, then code (goals for is not faked here). */
    fun ranked(s: List<SketchStanding>): List<SketchStanding> =
        s.sortedWith(compareByDescending<SketchStanding> { it.points }.thenByDescending { it.goalDiff }.thenBy { it.code })

    /** The fake match: 30 s on the clock, goals at these clock seconds elapsed (home = ours). */
    const val MATCH_SECONDS = 30.0
    val goals = listOf(SketchGoal(5.0, true), SketchGoal(11.0, false), SketchGoal(17.5, true), SketchGoal(24.0, true))
}

File: ios/Sources/Sketch/SketchData.swift
import Foundation

// THROWAWAY (SMASH-5): the motion sketch's fake data — a click-dummy for judging the 3D UI's
// feel on a phone, not the season (§11). Kit colours are invented here; the spec gives each club
// a primary and a secondary kit colour but does not yet name them.

enum SketchScreenID: String { case title, hub, match, result, coach }

struct SketchClub {
    let code: String
    let primary: Int
    let secondary: Int
}

struct SketchStanding {
    let code: String
    var played: Int
    var goalDiff: Int
    var points: Int
}

enum SketchData {
    static let clubs: [String: SketchClub] = [
        "MOS": SketchClub(code: "MOS", primary: 0x4F8A3C, secondary: 0xF2E6C9),
        "GLO": SketchClub(code: "GLO", primary: 0x6C4AB6, secondary: 0xFFD84D),
        "NEB": SketchClub(code: "NEB", primary: 0x1F3B73, secondary: 0x7FD1F5),
        "ROC": SketchClub(code: "ROC", primary: 0xE4572E, secondary: 0x2B2D42),
        "DUN": SketchClub(code: "DUN", primary: 0xE8A33D, secondary: 0x5A2E0E),
        "MIR": SketchClub(code: "MIR", primary: 0xC1476B, secondary: 0xF6E7D8),
        "GLW": SketchClub(code: "GLW", primary: 0x9CC7E0, secondary: 0x1D3557),
        "COR": SketchClub(code: "COR", primary: 0xFF7F66, secondary: 0x0F6E7A),
    ]
    static let player = "DUN"
    static let opponent = "NEB"

    /// The table after matchday 4.
    static let before: [SketchStanding] = [
        .init(code: "ROC", played: 4, goalDiff: 6, points: 10),
        .init(code: "NEB", played: 4, goalDiff: 4, points: 9),
        .init(code: "GLO", played: 4, goalDiff: 2, points: 7),
        .init(code: "DUN", played: 4, goalDiff: 1, points: 6),
        .init(code: "MOS", played: 4, goalDiff: 0, points: 5),
        .init(code: "COR", played: 4, goalDiff: -2, points: 4),
        .init(code: "MIR", played: 4, goalDiff: -4, points: 3),
        .init(code: "GLW", played: 4, goalDiff: -7, points: 1),
    ]

    /// The table after matchday 5: DUN 3–1 NEB, ROC 1–1 GLO, MOS 2–1 COR, MIR 1–0 GLW.
    static let after: [SketchStanding] = ranked([
        .init(code: "ROC", played: 5, goalDiff: 6, points: 11),
        .init(code: "NEB", played: 5, goalDiff: 2, points: 9),
        .init(code: "GLO", played: 5, goalDiff: 2, points: 8),
        .init(code: "DUN", played: 5, goalDiff: 3, points: 9),
        .init(code: "MOS", played: 5, goalDiff: 1, points: 8),
        .init(code: "COR", played: 5, goalDiff: -3, points: 4),
        .init(code: "MIR", played: 5, goalDiff: -3, points: 6),
        .init(code: "GLW", played: 5, goalDiff: -8, points: 1),
    ])

    /// §11.4's order: points, goal difference, then code (goals for is not faked here).
    static func ranked(_ s: [SketchStanding]) -> [SketchStanding] {
        s.sorted { ($0.points, $0.goalDiff, $1.code) > ($1.points, $1.goalDiff, $0.code) }
    }

    /// The fake match: 30 s on the clock, goals at these clock seconds elapsed (true = ours).
    static let matchSeconds = 30.0
    static let goals: [(at: Double, home: Bool)] = [(5, true), (11, false), (17.5, true), (24, true)]
}

/// Localized copy for the sketch (keys in Localizable.xcstrings).
func L(_ key: String.LocalizationValue) -> String { String(localized: key) }
